src/components/Physics.js

```diff
@@ -2,72 +2,135 @@ import Matter from 'matter-js';
 import Constants from './Constants.js';
 import Obstacle from './Obstacle.js';
 
-let jumps = 0;
 let obstacles = 0;
 let timer = 0;
+let frameCount = 0;
+let currentFrame = 1;
 
-export const randomNumber = (min, max) => {
-    return Math.floor(Math.random() * (max - min + 1) + min)
-}
-  
-export const generateObstacleHeight = () => {
-    let obstacleHeight = randomNumber(50, (Constants.MAX_HEIGHT / 8));
-    return obstacleHeight;
-}
 
-export const addObstacleToGame = (x, world, entities) => {
-    let obstacleHeight = generateObstacleHeight();
-    let width = Constants.OBSTACLE_WIDTH;
 
-    let obstacle = Matter.Bodies.rectangle(
-        x,
-        Constants.MAX_HEIGHT - (obstacleHeight / 2) - 175,
-        width,
-        obstacleHeight,
-        {isStatic: true}
+export const addObstacleToGame = (px, py, x, world, entities) => {
+    let rad1 = Matter.Common.random(30, 100);
+    let ox = x;
+    let oy = Constants.MAX_HEIGHT / 2;
+    let obstacle1 = Matter.Bodies.circle(
+        ox,
+        oy,
+        rad1,
+        {
+            isStatic: true,
+        }
     );
-
-    Matter.World.add(world, [obstacle]);
+    Matter.World.add(world, [obstacle1]);
+    // get force and direction to apply
+    let fx = px - ox;
+    let fy = py -oy;
+
+    // get rotation
+    let rotation = Math.asin(fy/fx);
+    // Matter.Body.rotate(obstacle1, rotation);
     entities["obstacle" + obstacles] = {
-        body: obstacle, size: [width, obstacleHeight], renderer: Obstacle
+        body: obstacle1, angle: rotation, fx: fx, fy: fy, rad: rad1, scoreClaimed: false, goingUp: true, renderer: Obstacle
     }
     obstacles += 1
 
 }
+export const collisionCheck = (otherBody, playerBody, dispatch) => {
+    let collision = Matter.SAT.collides(otherBody, playerBody);
+    if (collision.collided) {
+        dispatch({ type: "game-over" });
+    }
+}
 
-const Physics = (entities, { touches, time }) => {
+// export const addObstacleToGame = (x, world, entities) => {
```

Comment:
Is there a reason the commented code is checked in? If its important keep it, otherwise remove it